### PR TITLE
Nonpr 2525 sqs

### PR DESCRIPTION
**Pull request by Leeroyc720**

### CODE REVIEW

File: src/main/resources/logback.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
         <remoteHost>${LOGSTASH_URL}</remoteHost>
         <port>${LOGSTASH_PORT}</port>
         <encoder class="net.logstash.logback.encoder.LogstashEncoder">
-            <customFields>{"appname":"attachment", "env":"${ENV}"}</customFields>
+            <customFields>{"appname":"attachment-processor", "env":"${ENV}"}</customFields>
```
Comment (greg-gajda) [inline src/main/resources/logback.xml:7]: :+1:

File: build.sbt [hunk 2]
```diff
@@ -7,6 +7,7 @@ val logbackVersion = "1.2.10"
 val metricsVersion = "1.6.0"
 val jvmMetricsVersion = "3.0.2"
 val prometheusClientsVersion = "0.15.0"
+val sprayVersion = "1.3.6"
```
Comment (greg-gajda) [inline build.sbt:10]: you can consider io.circe as an alternative 
https://github.com/hmrc/nonrep-submission/blob/main/build.sbt#L68:L69

File: src/main/scala/uk/gov/hmrc/nonrep/attachment/package.scala [hunk 3]
```diff
@@ -21,4 +21,8 @@ package object attachment {
   case class AttachmentContent(info: AttachmentInfo, content: ByteString)
 
   case class ZipContent(info: AttachmentInfo, files: Seq[(String, Array[Byte])])
+
+  type SQSMessageParser = (String, String) => Option[List[AttachmentInfo]]
```
Comment (Leeroyc720) [inline src/main/scala/uk/gov/hmrc/nonrep/attachment/package.scala:25]: to remove this as its no longer used

File: src/main/scala/uk/gov/hmrc/nonrep/attachment/package.scala [hunk 4]
```diff
@@ -21,4 +21,8 @@ package object attachment {
...
+
+ val queueUrl = "https://sqs.eu-west-2.amazonaws.com/20304005294/MyQueue"  //example to be changed
```
Comment (greg-gajda) [inline src/main/scala/uk/gov/hmrc/nonrep/attachment/package.scala:27]: this is already defined in ServiceConfig as sqsTopicArn

File: src/main/scala/uk/gov/hmrc/nonrep/attachment/service/Queue.scala [hunk 5]
```diff
@@ -2,32 +2,79 @@ package uk.gov.hmrc.nonrep.attachment
...
+  val sourceSettings = SqsSourceSettings()
+    .withWaitTime(config.waitTime seconds)
+    .withMaxBufferSize(config.maxBufferSize)
+    .withMaxBatchSize(config.maxBatchSize)
+
+  def  Source() {
+        val messages: Future[immutable.Seq[Message]] =
+          SqsSource(
+            queueUrl,
+            SqsSourceSettings().withCloseOnEmptyReceive(true).withWaitTime(20.millis)
```
Comment (greg-gajda) [inline src/main/scala/uk/gov/hmrc/nonrep/attachment/service/Queue.scala:54]: these additional settings for withCloseOnEmptyReceive and withWaitTime should be easily moved to line 45 where sourceSettings are defined

File: src/main/scala/uk/gov/hmrc/nonrep/attachment/service/Queue.scala [hunk 6]
```diff
@@ -2,32 +2,79 @@ package uk.gov.hmrc.nonrep.attachment
...
 
   // make sure that SQS async client is created with important parameters taken from service config
 
   /*
    * all these implementations are likely to be replaced
    */
-  override def getMessages: Source[Message, NotUsed] = Source.empty
+  override def getMessages: Source[Message, NotUsed] = SqsSource(config.sqsTopicArn, sourceSettings)
 
   override def parseMessages: Flow[Message, AttachmentInfo, NotUsed] = Flow.fromFunction((m: Message) => AttachmentInfo(m.messageId(), ""))
```
Comment (greg-gajda) [inline src/main/scala/uk/gov/hmrc/nonrep/attachment/service/Queue.scala:74]: this method should be implemented

File: src/main/scala/uk/gov/hmrc/nonrep/attachment/service/Queue.scala [hunk 7]
```diff
@@ -2,32 +2,79 @@ package uk.gov.hmrc.nonrep.attachment
...
+
+
 
   // make sure that SQS async client is created with important parameters taken from service config
 
   /*
    * all these implementations are likely to be replaced
    */
-  override def getMessages: Source[Message, NotUsed] = Source.empty
+  override def getMessages: Source[Message, NotUsed] = SqsSource(config.sqsTopicArn, sourceSettings)
```
[the remark this answers is not in the document]
Comment (greg-gajda) [reply src/main/scala/uk/gov/hmrc/nonrep/attachment/service/Queue.scala:72]: Also, you may consider using https://doc.akka.io/docs/akka/current/stream/operators/RestartSource/withBackoff.html

**Comment by greg-gajda** [comment]
Finally, different implementation was chosen

**Pull request closed without merging**